subsys/bluetooth/mesh/crypto_mtls.c

```diff
@@ -0,0 +1,643 @@
+		fcs = crc_table[fcs ^ *data++];
+	}
+
+	BT_DBG("fcs 0x%02x", 0xff - fcs);
+
+	return 0xff - fcs;
+}
+
+bool bt_mesh_fcs_check(struct net_buf_simple *buf, uint8_t received_fcs)
+{
+	const uint8_t *data = buf->data;
+	uint16_t data_len = buf->len;
+	uint8_t fcs = 0xff;
+
+	while (data_len--) {
+		fcs = crc_table[fcs ^ *data++];
+	}
+
+	return crc_table[fcs ^ received_fcs] == 0xcf;
+}
+
+int bt_mesh_virtual_addr(const uint8_t virtual_label[16], uint16_t *addr)
+{
+	uint8_t salt[16];
+	uint8_t tmp[16];
+	int err;
+
+	err = bt_mesh_s1("vtad", salt);
+	if (err) {
+		return err;
+	}
+
+	err = bt_mesh_aes_cmac_one(salt, virtual_label, 16, tmp);
+	if (err) {
+		return err;
+	}
+
+	*addr = (sys_get_be16(&tmp[14]) & 0x3fff) | 0x8000;
+
+	return 0;
+}
+
+int bt_mesh_prov_salt(const uint8_t conf_salt[16], const uint8_t prov_rand[16],
+		      const uint8_t dev_rand[16], uint8_t prov_salt[16])
+{
+	const uint8_t prov_salt_key[16] = { 0 };
+	struct bt_mesh_sg sg[] = {
+		{ conf_salt, 16 },
+		{ prov_rand, 16 },
+		{ dev_rand, 16 },
+	};
+
+	return bt_mesh_aes_cmac(prov_salt_key, sg, ARRAY_SIZE(sg), prov_salt);
+}
+
+int bt_mesh_prov_conf_salt(const uint8_t conf_inputs[145], uint8_t salt[16])
+{
+	const uint8_t conf_salt_key[16] = { 0 };
+
+	return bt_mesh_aes_cmac_one(conf_salt_key, conf_inputs, 145, salt);
+}
+
+int bt_mesh_prov_conf_key(const uint8_t dhkey[32], const uint8_t conf_salt[16],
+			  uint8_t conf_key[16])
+{
+	return bt_mesh_k1(dhkey, 32, conf_salt, "prck", conf_key);
+}
+
+int bt_mesh_prov_conf(const uint8_t conf_key[16], const uint8_t rand[16],
+		      const uint8_t auth[16], uint8_t conf[16])
+{
+	struct bt_mesh_sg sg[] = { { rand, 16 }, { auth, 16 } };
+
+	BT_DBG("ConfirmationKey %s", bt_hex(conf_key, 16));
+	BT_DBG("RandomDevice %s", bt_hex(rand, 16));
+	BT_DBG("AuthValue %s", bt_hex(auth, 16));
+
+	return bt_mesh_aes_cmac(conf_key, sg, ARRAY_SIZE(sg), conf);
+}
+
+int bt_mesh_prov_decrypt(const uint8_t key[16], uint8_t nonce[13],
+			 const uint8_t data[25 + 8], uint8_t out[25])
+{
+	return bt_ccm_decrypt(key, nonce, data, 25, NULL, 0, out, 8);
+}
+
+int bt_mesh_prov_encrypt(const uint8_t key[16], uint8_t nonce[13],
+			 const uint8_t data[25], uint8_t out[25 + 8])
+{
+	return bt_ccm_encrypt(key, nonce, data, 25, NULL, 0, out, 8);
+}
+
+int bt_mesh_beacon_auth(const uint8_t beacon_key[16], uint8_t flags,
+			const uint8_t net_id[8], uint32_t iv_index,
+			uint8_t auth[8])
+{
+	uint8_t msg[13], tmp[16];
+	int err;
+
+	BT_DBG("BeaconKey %s", bt_hex(beacon_key, 16));
+	BT_DBG("NetId %s", bt_hex(net_id, 8));
+	BT_DBG("IV Index 0x%08x", iv_index);
+
+	msg[0] = flags;
+	memcpy(&msg[1], net_id, 8);
+	sys_put_be32(iv_index, &msg[9]);
+
+	BT_DBG("BeaconMsg %s", bt_hex(msg, sizeof(msg)));
+
+	err = bt_mesh_aes_cmac_one(beacon_key, msg, sizeof(msg), tmp);
+	if (!err) {
+		memcpy(auth, tmp, 8);
+	}
+
+	return err;
+}
+
+int bt_mesh_dhkey_gen(const uint8_t *pub_key, const uint8_t *priv_key, uint8_t *dhkey)
+{
+	return -ENOSYS;
+}
+
```

Comment:
for some reason after migration on Zephyr's configuration mbedTLS mandated to have implementation of this function. 
Another option is to have internal entropy that is based on sha256 or sha512. I wasn't able to ron mbtls with entropy_c option. 
Probably it requires more config options that the checker cannot catch?